### PR TITLE
Python3 migration

### PR DESCRIPTION
**Pull request by jnburchett**
Getting with the times...

### CODE REVIEW

File: joebvp/VPmeasure.py [hunk 1]
```diff
@@ -25,16 +26,22 @@
 from linetools.spectra.io import readspec
 import numpy as np
 from astropy.constants import c
-
+try:
+    from importlib import reload
+except:
+    pass
```
Comment (ntejos) [inline joebvp/VPmeasure.py:32]: Is this necessary?

File: joebvp/atomicdata/__init__.py [hunk 2]
```diff
@@ -17,6 +23,14 @@
...
 vernp=jbg.arrfromcol(vernerlist,7)
 
+starts = (0,10,18,21,24,27,30,40)
+verntab = Table.read(jbvp_path+'/atomicdata/verner6.txt',
+                     format='ascii.fixed_width_no_header',
+                     col_starts=starts,fill_values=('','0'),
+                     names=['lam','ion','Z','nume','gl','gu','fosc','P'])
+verntab['fosc'].astype(float)
+vernion=verntab['ion']
+
```
Comment (ntejos) [inline joebvp/atomicdata/__init__.py:33]: I think Verner table is included in LineList 'ISM'

File: joebvp/joebgoodies.py [hunk 3]
```diff
@@ -261,28 +194,14 @@ def wherebetween(testval,lim1,lim2):
...
-# 	return 2.356e5*dist**2*S*dv
-# '''
+
+### Estimate total H I mass given flat line profile
+def himass(dist,S,dv):
+ 	'''
+ 	Units: [dist] = Mpc, [S] = Jy, [dv] = km/s
+ 	'''
+ 	return 2.356e5*dist**2*S*dv
+
```
Comment (ntejos) [inline joebvp/joebgoodies.py:198]: Why is this here?

File: joebvp/makevoigt.py [hunk 4]
```diff
@@ -95,18 +97,27 @@ def get_lsfs():
             cfg.lsfs.append(lsf['kernel'])
             break
         else:
-
-            lamobs=np.median(cfg.wave[fg])
+            try:
+                lamobs=np.median(cfg.wave[fg])
+            except:
+                import pdb; pdb.set_trace()
```
Comment (ntejos) [inline joebvp/makevoigt.py]: clean this?

File: joebvp/makevoigt.py [hunk 5]
```diff
@@ -95,18 +97,27 @@ def get_lsfs():
...
             lsfmatch = jbg.wherebetween(lamobs, cfg.lsfranges[:, 0], cfg.lsfranges[:, 1])
+            try:
+                lfg = len(fg)
+            except:
+                import pdb; pdb.set_trace()
```
Comment (ntejos) [inline joebvp/makevoigt.py]: and this?

File: joebvp/makevoigt.py [hunk 6]
```diff
@@ -95,18 +97,27 @@ def get_lsfs():
...
                     import pdb;pdb.set_trace()
                     # NT: this could happen when pixels with S/N<0 exist, i.e. pixels where flux is <0 (e.g. black lines).
                     # For this reason, is better to remove the option to eliminate pixels based on S/N.
-                n_more_side = (10 - len(fg))/2 + 1
-                inds_right = [np.max(fg) + ii + 1 for ii in range(n_more_side)]
+                n_more_side = int(np.ceil((10 - len(fg))/2 + 1))
+                try:
+                    inds_right = [np.max(fg) + ii + 1 for ii in range(n_more_side)]
+                except:
+                    import pdb; pdb.set_trace()
```
Comment (ntejos) [inline joebvp/makevoigt.py]: ditto

File: joebvp/makevoigt.py [hunk 7]
```diff
@@ -183,15 +197,18 @@ def convolvecos(wave,profile,lines,zs):
...
             break
         else:
 
-            lsfwidth=len(ll)/2+1
-            paddedprof = np.insert(profile[ll], 0, [1.] * lsfwidth)
+            lsfwidth=int(np.ceil(len(ll)/2+1))
+            try:
+                paddedprof = np.insert(profile[ll], 0, [1.] * lsfwidth)
+            except:
+                import pdb; pdb.set_trace()
```
Comment (ntejos) [inline joebvp/makevoigt.py]: ditto

File: joebvp/utils.py [hunk 8]
```diff
@@ -1,14 +1,16 @@
...
+from joebvp import makevoigt
+from joebvp import atomicdata
 from linetools.spectra.io import readspec
 from linetools.spectra.xspectrum1d import XSpectrum1D
+from linetools.lists.linelist import LineList
 from astropy.table import Table,vstack
 from astropy.io import ascii
 try:
-    import joebvp_cfg as cfg
+    from joebvp import joebvp_cfg as cfg
```
Comment (ntejos) [inline joebvp/utils.py]: This should be removed, because the idea was to import a local configuration file.

**Review by ntejos — COMMENTED** [review]
Looks good, please review my comments below.

File: joebvp/VPmeasure.py (hunk 1 above)
Comment (jnburchett) [reply joebvp/VPmeasure.py:32]: I reload the cfg.py file is to clear the LSFs loaded for each spectral chunk fitted, and python 3 doesn't natively have a reload() function.  The `try except` business is because python 2 doesn't have the importlib package.  This allows some backward compatibility.

File: joebvp/joebgoodies.py (hunk 3 above)
Comment (jnburchett) [reply joebvp/joebgoodies.py:198]: Holdover from the code from which I had pasted a number of these routines.  Removed now.

File: joebvp/atomicdata/__init__.py (hunk 2 above)
Comment (jnburchett) [reply joebvp/atomicdata/__init__.py:33]: True, and I just verified that linetools is being used for all instances where we are retrieving actual atomic data.  The one thing we still look up in the Verner table is the species name, e.g., 'H I' where the LineList class doesn't simply serve up the species name.  I'm inclined to deprecate this at some point and add a 'species' attribute to LineList.  Gonna punt for now though.

File: joebvp/makevoigt.py (hunk 4 above)
Comment (jnburchett) [reply joebvp/makevoigt.py]: Lots of debugging involved in getting the LSF issues worked out...cleaning this stuff up now!

File: joebvp/makevoigt.py (hunk 5 above)
Comment (jnburchett) [reply joebvp/makevoigt.py]: yep

File: joebvp/makevoigt.py (hunk 6 above)
Comment (jnburchett) [reply joebvp/makevoigt.py]: done

File: joebvp/makevoigt.py (hunk 7 above)
Comment (jnburchett) [reply joebvp/makevoigt.py]: yep

File: joebvp/utils.py (hunk 8 above)
Comment (jnburchett) [reply joebvp/utils.py]: Should be 
`try:
    import joebvp_cfg`
Right?

File: joebvp/atomicdata/__init__.py (hunk 2 above)
Comment (ntejos) [reply joebvp/atomicdata/__init__.py:33]: Do not follow, you can get the species names in LineList object by removing the second part of the name string of the ion transition...

File: joebvp/utils.py (hunk 8 above)
Comment (jnburchett) [reply joebvp/utils.py]: Corrected

**Comment by jnburchett** [comment]
Okay, @ntejos , could you merge if all looks okay?

**Comment by ntejos** [comment]
Looks good, but there are some conflicts that need to be addressed before merging. Want me to resolve these?

File: joebvp/atomicdata/__init__.py (hunk 2 above)
Comment (jnburchett) [reply joebvp/atomicdata/__init__.py:33]: This is true, but the spacing between, e.g., 'H I' and 'SiIII' as they appear in our output tables.  Out of fear of breaking functions that look up data by these species names, I'm punting here.
